tests/unit/test_proxysql_manager.py

```diff
@@ -53,3 +54,103 @@ def test__deregister_backend_does_not_delete_the_backend_if_not_registered():
 
     proxysql_man.deregister_backend(backend.hostgroup_id, backend.hostname,
                                     backend.port)
+
+
+def test_register_backend():
+    proxysql_man = ProxySQLManager('127.0.0.1', 6032, 'username', 'password')
+    backend = ProxySQLMySQLBackend({
+        'hostgroup_id': 10,
+        'hostname': 'dummy_host',
+        'port': 3306
+    })
+
+    (allow(proxysql_man)
+        .get_connection
+        .and_return(MagicMock()))
+
+    (expect(proxysql_man)
+        .insert_or_update_mysql_backend
+        .once())
+
+    proxysql_man.register_backend(backend.hostgroup_id, backend.hostname, backend.port)
+
+
+def test_update_mysql_backend_status_if_backend_registered():
+    proxysql_man = ProxySQLManager('127.0.0.1', 6032, 'username', 'password')
+    backend = ProxySQLMySQLBackend({
+        'hostgroup_id': 10,
+        'hostname': 'dummy_host',
+        'port': 3306
+    })
+    (allow(proxysql_man)
+        .get_connection
+        .and_return(MagicMock()))
+
+    (allow(proxysql_man)
+        .is_mysql_backend_registered
+        .and_return(True))
+
+    (expect(proxysql_man)
+        .insert_or_update_mysql_backend
+        .once())
+
+    proxysql_man.update_mysql_backend_status(backend.hostgroup_id, backend.hostname, backend.port, "dummy_status")
+
+
+def test_update_mysql_backend_status_if_backend_not_registered():
+    proxysql_man = ProxySQLManager('127.0.0.1', 6032, 'username', 'password')
+    backend = ProxySQLMySQLBackend({
+        'hostgroup_id': 10,
+        'hostname': 'dummy_host',
+        'port': 3306
+    })
+    (allow(proxysql_man)
+        .get_connection
+        .and_return(MagicMock()))
+
+    (allow(proxysql_man)
+        .is_mysql_backend_registered
+        .and_return(False))
+
+    (expect(proxysql_man)
+        .insert_or_update_mysql_backend
+        .never())
+
+    with pytest.raises(ProxySQLMySQLBackendUnregistered):
+        proxysql_man.update_mysql_backend_status(backend.hostgroup_id, backend.hostname, backend.port, "dummy_status")
+
+
+def test_register_mysql_user_if_registered():
```

Comment:
Rename the function to `test__register_mysql_user_does_not_update_if_already_registered`